### PR TITLE
[main] Add compressed firmware support

### PR DESCRIPTION
**Pull request by vinceaperri**
###### Merge Checklist  
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

The QLogic NIC firmware in linux-firmware is compressed, but the kernel was not able to load it. This PR add that ability.

###### Change Log  

- Four config options were enabled for the kernel:
    - FW_LOADER_PAGED_BUF
    - FW_LOADER_USER_HELPER
    - FW_LOADER_COMPRESS
- Changed packages:
    - kernel
    - kernel-rt
    - kernel-headers (version bump only)

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build id: xxxxx

### CODE REVIEW

File: SPECS/kernel/config [hunk 1]
```diff
@@ -1870,9 +1870,11 @@ CONFIG_PREVENT_FIRMWARE_BUILD=y
 # Firmware loader
 #
 CONFIG_FW_LOADER=y
+CONFIG_FW_LOADER_PAGED_BUF=y
```
Comment (Camelron) [inline SPECS/kernel/config:1873]: Can you add these config changes to the ARM config as well? SPECS/kernel/config_aarch64

File: toolkit/resources/manifests/package/pkggen_core_aarch64.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 filesystem-1.1-8.cm2.aarch64.rpm
-kernel-headers-5.15.2.1-5.cm2.noarch.rpm
+kernel-headers-5.15.2.1-6.cm2.noarch.rpm
```
Comment (Camelron) [inline toolkit/resources/manifests/package/pkggen_core_aarch64.txt:2]: You're going to need to rebase onto the recently merged kernel update https://github.com/microsoft/CBL-Mariner/pull/2104

**Review by romoh — APPROVED** [review]
Apart from Camelron comments, change looks good.

**Review by romoh — APPROVED** [review]

**Review by rlmenge — APPROVED** [review]
LGTM

**Review by christopherco — APPROVED** [review]

**Review by Camelron — APPROVED** [review]
LGTM

**Pull request merged**